telethon/extensions/binaryreader.py

```diff
@@ -55,7 +66,12 @@ def read_large_int(self, bits, signed=True):
 
     def read(self, length=-1):
         """Read the given amount of bytes, or -1 to read all remaining."""
-        result = self.stream.read(length)
+        if length >= 0:
+            result = self.stream[self.position:self.position + length]
```

Comment:
Would it make sense to check we don't overshoot?

Reply:
It is already validated below. Just wanted to minimize changes